### PR TITLE
feat: Implement append mode for a region

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
- https://github.com/GreptimeTeam/greptimedb/issues/3480

## What's changed and what's your intention?
This PR implements append mode for a mito region.
- By default, `append_mode` is false and the engine removes duplicate rows.
- If `append_mode` is true, the engine keeps all duplicate rows.

It adds a `dedup` flag to the `MergeReader`. The `MergeReader` will keep rows with duplicate keys but still yield rows in sorted order.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/mito2/src/read/merge.rs [hunk 1]
```diff
@@ -176,36 +181,58 @@ impl MergeReader {
...
+            duplicate_pos
+        };
+        Self::maybe_output_batch(
+            top.slice(0, output_end),
+            &mut self.output_batch,
+            self.dedup,
+            &mut self.metrics,
+        )?;
+        top_node.skip_rows(output_end, &mut self.metrics).await?;
+        self.reheap(top_node)
```
Comment (evenyag) [inline src/mito2/src/read/merge.rs:235]: This is similar to the memtable [merger](https://github.com/GreptimeTeam/greptimedb/blob/175929426a8f06c03a087b1d3b380b9f7af2a474/src/mito2/src/memtable/partition_tree/merger.rs#L113-L122)

**Review by fengjiachun — APPROVED** [review]
LGTM

**Review by waynexia — COMMENTED** [review]
How to set this mode? Rest LGTM

**Comment by evenyag** [comment]
> How to set this mode? Rest LGTM

I'll expose this to table options in the next PR.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**